scipy/stats/_distn_infrastructure.py

```diff
@@ -1892,49 +1907,39 @@ def nnlf(self, theta, x):
         This is ``-sum(log pdf(x, theta), axis=0)`` where `theta` are the
         parameters (including loc and scale).
         '''
-        try:
-            loc = theta[-2]
-            scale = theta[-1]
-            args = tuple(theta[:-2])
-        except IndexError:
-            raise ValueError("Not enough input arguments.")
+        loc, scale, args = self._unpack_loc_scale(theta)
         if not self._argcheck(*args) or scale <= 0:
             return inf
         x = asarray((x-loc) / scale)
-        cond0 = (x <= self.a) | (self.b <= x)
-        if np.any(cond0):
+        n_log_scale = len(x) * log(scale)
+        if np.any(~self._support_mask(x)):
             return inf
-        else:
-            N = len(x)
-            return self._nnlf(x, *args) + N * log(scale)
+        return self._nnlf(x, *args) + n_log_scale
+
+    def _nnlf_and_penalty(self, x, args):
+        cond0 = ~self._support_mask(x)
+        n_bad = sum(cond0)
+        if n_bad > 0:
+            x = argsreduce(~cond0, x)[0]
+        logpdf = self._logpdf(x, *args)
+        finite_logpdf = np.isfinite(logpdf)
+        n_bad += np.sum(1 - finite_logpdf, axis=0)
```

Comment:
Summing a bool array directly seems to work, is there something more to it (overflows etc)? 

```
In [24]: a = np.ones(4, dtype=bool)

In [25]: np.sum(a)
Out[25]: 4
```

This is a minor point and I'm not going to press it anyway.